servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/GrpcUtils.java

```diff
@@ -167,18 +165,31 @@ static HttpResponse newResponse(final HttpResponseFactory responseFactory,
         return response;
     }
 
-    static HttpResponse newErrorResponse(final HttpResponseFactory responseFactory,
-                                         @Nullable final GrpcServiceContext context,
-                                         final Throwable cause, final BufferAllocator allocator) {
-        HttpResponse response = newResponse(responseFactory, context, allocator);
-        setStatus(response.trailers(), cause, allocator);
+    static StreamingHttpResponse newTrailersOnlyErrorResponse(final StreamingHttpResponseFactory responseFactory,
+                                                     final GrpcStatus status,
+                                                     final BufferAllocator allocator) {
+        final StreamingHttpResponse response = responseFactory.ok();
+        initResponse(response, null);
+        setStatus(response.headers(), status, null, allocator);
+        return response;
+    }
+
+    static HttpResponse newTrailersOnlyErrorResponse(final HttpResponseFactory responseFactory,
+                                                     @Nullable final GrpcServiceContext context,
+                                                     final Throwable cause, final BufferAllocator allocator) {
+        final HttpResponse response = responseFactory.ok();
+        initResponse(response, context);
```

Comment:
According to the spec [1] the server MAY send the Message-Accept-Encoding to the client. Its mostly useful for cases where an incoming encoding isn't supported so you advertise to the peer which ones are supported. We can tune it down and only produce it in case of a compression error. Thoughts?

[2] https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md#responses
> Response-Headers → HTTP-Status [Message-Encoding] [Message-Accept-Encoding] Content-Type *Custom-Metadata

Reply:
I missed that part, thanks for pointing. Let's keep as-is for now.